apps/web/app/(navigation)/LanguageSelector.tsx

```diff
@@ -0,0 +1,75 @@
+import * as Select from "@radix-ui/react-select";
+
+import { Flag } from "ui";
+import {
+  SelectContentStyled,
+  SelectItemStyled,
+  SelectTriggerStyled,
+} from "./LanguageSelectorStyled";
+import { languageAtom, languages } from "app/store";
+import { useSetAtom, useAtomValue } from "jotai";
+import { useHasScrollBar } from "lib/hooks/useHasScrollBar";
+
+export const LanguageSelector = () => {
+  const { hasScrollbar } = useHasScrollBar();
+  const setLanguage = useSetAtom(languageAtom);
+  const language = useAtomValue(languageAtom);
+
+  const changeLanguage = (lang: string) => {
+    setLanguage(lang as languages);
+    localStorage.setItem("lang", lang);
+  };
```

Comment:
I explained why this suggestion occured on `apps/web/app/layout.tsx` file


```suggestion
  const changeLanguage = (lang: string) => {
    setLanguage(lang as languages);
    localStorage.setItem("lang", lang);
    
      useEffect(() => {
    //localStorage.getItem("lang") works as well
    setLanguage(localStorage.lang as Language);
  }, [setLanguage]);
    
  };
```